File: javaSpringShip/src/main/java/pl/battleships/javaspringship/api/InformationController.java
package pl.battleships.javaspringship.api;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.CrossOrigin;
import pl.battleships.api.InformationApi;
import pl.battleships.api.dto.InfoDto;

import java.util.List;

@Controller
@RequiredArgsConstructor
@Slf4j
@CrossOrigin
public class InformationController implements InformationApi {

    @Override
    public ResponseEntity<InfoDto> info() {
        return ResponseEntity.ok(new InfoDto().name("javaSpring").details(List.of("java", "spring")));
    }
}

File: kotlinSpringShip/src/main/kotlin/pl/battleships/kotlinspringship/api/InformationController.kt
package pl.battleships.kotlinspringship.api

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.CrossOrigin
import org.springframework.web.bind.annotation.RestController
import pl.battleships.api.InformationApi
import pl.battleships.api.dto.InfoDto

@RestController
@CrossOrigin
class InformationController : InformationApi {

    override fun info(): ResponseEntity<InfoDto> {
        return ResponseEntity.ok(InfoDto("kotlinSpring", listOf("kotlin", "spring")))
    }
}